addons/mail/models/mail_channel.py

```diff
@@ -717,7 +727,14 @@ def _execute_channel_pin(self, pinned=False):
         self.ensure_one()
         channel_partners = self.env['mail.channel.partner'].search(
             [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', self.id), ('is_pinned', '!=', pinned)])
-        self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', self.env.user.partner_id.id), self.channel_info('unsubscribe' if not pinned else False)[0])
+        if not pinned:
```

Comment:
I don't think this condition is correct, it used to change the "info", not completely prevent the notification.